### PR TITLE
Woo: Add API call to delete product variation

### PR DESCRIPTION
**Pull request by coderkevin**
This adds the code necessary to both delete the product variation via
the API, and remove it from the local state.

To test: `npm test`

### CODE REVIEW

**Comment by coderkevin** [comment]
Rebased.

File: client/extensions/woocommerce/state/data-layer/product-variations/index.js [hunk 1]
```diff
@@ -87,3 +89,26 @@ export function handleProductVariationUpdate( store, action ) {
...
+			message: 'Attempting to delete a variation without a valid id.',
+			variationId,
+		} ) );
+		return;
+	}
+
+	const updatedAction = ( dispatch, getState, data ) => {
+		dispatch( productVariationUpdated( siteId, undefined, action ) );
+
+		const props = { productId: productId, sentData: { id: variationId }, receivedData: data };
```
Comment (justinshreve) [inline client/extensions/woocommerce/state/data-layer/product-variations/index.js]: `productId: productId` can just become `productId`

File: client/extensions/woocommerce/state/sites/product-variations/actions.js [hunk 2]
```diff
@@ -72,6 +73,27 @@ export function updateProductVariation( siteId, productId, variation, successAct
 }
 
 /**
+ * Action Creator: Delete an existing product variation.
+ *
+ * @param {Number} siteId The id of the site in which the variation's product exists.
+ * @param {Number} productId The id of the product for which this variation.
```
Comment (justinshreve) [inline client/extensions/woocommerce/state/sites/product-variations/actions.js]: for which this variation belongs

Comment (coderkevin) [inline client/extensions/woocommerce/state/sites/product-variations/actions.js]: Thanks!

File: client/extensions/woocommerce/state/data-layer/product-variations/index.js (hunk 1 above)
Comment (coderkevin) [inline client/extensions/woocommerce/state/data-layer/product-variations/index.js]: Thanks!

**Pull request merged**